pkg/steps/release/snapshot.go

```diff
@@ -64,19 +70,92 @@ func snapshotStream(ctx context.Context, client loggingclient.LoggingClient, sou
 		snapshot.ObjectMeta.Annotations[releaseConfigAnnotation] = raw
 	}
 	for _, tag := range source.Status.Tags {
-		if valid, _ := utils.FindStatusTag(source, tag.Tag); valid != nil {
-			snapshot.Spec.Tags = append(snapshot.Spec.Tags, imagev1.TagReference{
-				Name:            tag.Tag,
-				From:            valid,
-				ReferencePolicy: imagev1.TagReferencePolicy{Type: imagev1.LocalTagReferencePolicy},
-			})
-		}
+		snapshot.Spec.Tags = append(snapshot.Spec.Tags, imagev1.TagReference{
+			Name: tag.Tag,
+			From: &coreapi.ObjectReference{
+				Kind: "DockerImage",
+				Name: api.QCIAPPCIImage(api.ImageStreamTagReference{Namespace: sourceNamespace, Name: sourceName, Tag: tag.Tag}),
+			},
+			ImportPolicy:    imagev1.TagImportPolicy{ImportMode: imagev1.ImportModePreserveOriginal},
+			ReferencePolicy: imagev1.TagReferencePolicy{Type: imagev1.LocalTagReferencePolicy},
+		})
 	}
 	// the Create call mutates the input object, so we need to copy it before returning
 	created := snapshot.DeepCopy()
 	if err := client.Create(ctx, created); err != nil && !kerrors.IsAlreadyExists(err) {
 		return nil, nil, fmt.Errorf("could not create snapshot imagestream %s/%s for release %s: %w", sourceNamespace, sourceName, targetRelease, err)
 	}
+	begin := time.Now()
+	logrus.Infof("Waiting to import tags on imagestream %s/%s ...", created.Namespace, created.Name)
+	for i, tag := range created.Spec.Tags {
```

Comment:
Do we care to do this in serial only?

Reply:
The imports takes places in the OpenShift controllers (I believe they are concurrent).
Here we check if they have done the job before moving forward.
From the tests and rehearsals, the controller update the status on the stream at the same time.
Checking the tags one after another sounds reasonable to me here.
We have other places with a similar checking implementation already.
We might switch to the watch mode in the future if it creates to much workload on the api server by periodic `get`.

Reply:
Yes, this can be tuned in a follow-up. However, the iteration will wait for the `PollUntilContextTimeout` to stop and then continue. If there is a problem with one image, it will get stuck until timeout either way.